plugins/database/redshift/redshift.go

```diff
@@ -229,132 +172,180 @@ func (r *RedShift) CreateUser(ctx context.Context, statements dbplugin.Statement
 				"expiration": expirationStr,
 			}
 			if err := dbtxn.ExecuteTxQuery(ctx, tx, m, query); err != nil {
-				return "", "", err
+				return dbplugin.NewUserResponse{}, err
 			}
 		}
 	}
 
 	// Commit the transaction
 	if err := tx.Commit(); err != nil {
-		return "", "", err
+		return dbplugin.NewUserResponse{}, err
 	}
-	return username, password, nil
+	return dbplugin.NewUserResponse{
+		Username: username,
+	}, nil
 }
 
-func (r *RedShift) RenewUser(ctx context.Context, statements dbplugin.Statements, username string, expiration time.Time) error {
+// UpdateUser can update the expiration or the password of a user, or both.
+// The updates all happen in a single transaction, so they will either all
+// succeed or all fail.
+// Both updates support both default and custom statements.
+func (r *RedShift) UpdateUser(ctx context.Context, req dbplugin.UpdateUserRequest) (dbplugin.UpdateUserResponse, error) {
+	if req.Password == nil && req.Expiration == nil {
+		return dbplugin.UpdateUserResponse{}, nil
+	}
+
 	r.Lock()
 	defer r.Unlock()
 
-	statements = dbutil.StatementCompatibilityHelper(statements)
-
-	renewStmts := statements.Renewal
-	if len(renewStmts) == 0 {
-		renewStmts = []string{defaultRenewSQL}
-	}
-
 	db, err := r.getConnection(ctx)
 	if err != nil {
-		return err
+		return dbplugin.UpdateUserResponse{}, err
 	}
 	defer db.Close()
 
 	tx, err := db.BeginTx(ctx, nil)
 	if err != nil {
-		return err
+		return dbplugin.UpdateUserResponse{}, err
 	}
 	defer func() {
 		tx.Rollback()
 	}()
 
-	expirationStr, err := r.GenerateExpiration(expiration)
-	if err != nil {
-		return err
-	}
+	if req.Expiration != nil {
+		renewStmts := req.Expiration.Statements
+		if len(renewStmts.Commands) == 0 {
+			renewStmts.Commands = []string{defaultRenewSQL}
+		}
 
-	for _, stmt := range renewStmts {
-		for _, query := range strutil.ParseArbitraryStringSlice(stmt, ";") {
-			query = strings.TrimSpace(query)
-			if len(query) == 0 {
-				continue
+		expirationStr := req.Expiration.NewExpiration.UTC().Format("2006-01-02 15:04:05")
+
+		for _, stmt := range renewStmts.Commands {
+			for _, query := range strutil.ParseArbitraryStringSlice(stmt, ";") {
+				query = strings.TrimSpace(query)
+				if len(query) == 0 {
+					continue
+				}
+
+				m := map[string]string{
+					"name":       req.Username,
+					"expiration": expirationStr,
+				}
+				if err := dbtxn.ExecuteTxQuery(ctx, tx, m, query); err != nil {
+					return dbplugin.UpdateUserResponse{}, err
+				}
 			}
+		}
+	}
 
-			m := map[string]string{
-				"name":       username,
-				"expiration": expirationStr,
-			}
-			if err := dbtxn.ExecuteTxQuery(ctx, tx, m, query); err != nil {
-				return err
+	if req.Password != nil {
+		username := req.Username
+		password := req.Password.NewPassword
+		if username == "" || password == "" {
+			return dbplugin.UpdateUserResponse{}, errors.New("must provide both username and password")
+		}
+
+		// Check if the role exists
+		var exists bool
```

Comment:
This is never referenced. Shouldn't we be checking if this is false and returning an error?